src/common.cpp

```diff
@@ -926,23 +925,32 @@ bool load_f64(PyObject *o, uint8_t flags, double *out) noexcept {
 }
 
 bool load_f32(PyObject *o, uint8_t flags, float *out) noexcept {
+    static_assert(std::numeric_limits<float>::is_iec559);
```

Comment:
I think you can remove these. Nanobind is specifically scoped to architectures and compielrs that satisfy these assertions.